File: examples/cpp/my-app/src/models/Task.cpp
#include "models/Task.h"
#include "utils/DateUtils.h"

namespace models {

Task::Task(int id, const std::string& title, const std::string& description, int projectId)
    : id(id), title(title), description(description), projectId(projectId),
      status(TaskStatus::TODO), priority(TaskPriority::MEDIUM),
      assigneeId(-1), createdAt(utils::DateUtils::now()), dueDate(0) {}

int Task::getId() const {
    return id;
}

std::string Task::getTitle() const {
    return title;
}

std::string Task::getDescription() const {
    return description;
}

TaskStatus Task::getStatus() const {
    return status;
}

TaskPriority Task::getPriority() const {
    return priority;
}

int Task::getProjectId() const {
    return projectId;
}

int Task::getAssigneeId() const {
    return assigneeId;
}

void Task::setTitle(const std::string& newTitle) {
    title = newTitle;
}

void Task::setDescription(const std::string& newDescription) {
    description = newDescription;
}

void Task::setStatus(TaskStatus newStatus) {
    status = newStatus;
}

void Task::setPriority(TaskPriority newPriority) {
    priority = newPriority;
}

void Task::setAssignee(int userId) {
    assigneeId = userId;
}

void Task::setDueDate(std::time_t date) {
    dueDate = date;
}

void Task::addComment(const Comment& comment) {
    comments.push_back(comment);
}

} // namespace models

File: examples/python/my-app/src/models/task.py
"""Task model."""

from dataclasses import dataclass
from datetime import datetime
from typing import Optional
from enum import Enum

class TaskStatus(Enum):
    """Task status enumeration."""
    TODO = "todo"
    IN_PROGRESS = "in_progress"
    DONE = "done"
    CANCELLED = "cancelled"

class TaskPriority(Enum):
    """Task priority enumeration."""
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    URGENT = "urgent"

@dataclass
class Task:
    """Task entity."""
    
    id: int
    title: str
    description: str
    status: TaskStatus
    priority: TaskPriority
    assignee_id: int
    project_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    due_date: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    
    def is_overdue(self) -> bool:
        """Check if task is overdue."""
        if not self.due_date:
            return False
        return self.status != TaskStatus.DONE and datetime.now() > self.due_date
    
    def complete(self) -> None:
        """Mark task as completed."""
        self.status = TaskStatus.DONE
        self.completed_at = datetime.now()
        self.updated_at = datetime.now()
    
    def start(self) -> None:
        """Start working on task."""
        self.status = TaskStatus.IN_PROGRESS
        self.updated_at = datetime.now()
    
    def cancel(self) -> None:
        """Cancel the task."""
        self.status = TaskStatus.CANCELLED
        self.updated_at = datetime.now()
